hir/src/pass/manager.rs

```diff
@@ -29,11 +38,58 @@ pub enum PassDisplayMode {
 
 // TODO(pauls)
 #[allow(unused)]
+#[derive(Default, Debug)]
 pub struct IRPrintingConfig {
-    print_module_scope: bool,
-    print_after_only_on_change: bool,
-    print_after_only_on_failure: bool,
-    flags: OpPrintingFlags,
+    pub print_module_scope: bool,
+    pub print_after_only_on_failure: bool,
+    // NOTE: Taken from the Options struct
+    pub print_ir_after_all: bool,
+    pub print_ir_after_pass: SmallVec<[String; 1]>,
+    pub print_ir_after_modified: bool,
+    pub flags: OpPrintingFlags,
+}
+
+impl TryFrom<&Options> for IRPrintingConfig {
+    type Error = Report;
+
+    fn try_from(options: &Options) -> Result<Self, Self::Error> {
+        let available_passes = [
```

Comment:
I would remove this bit for now, as the set of passes is open - what dictates whether the pass name is "invalid" is whether or not such a pass has been registered with a specific `PassManager` instance (and more precisely, somewhere in the pipeline being managed by that manager).

So if we want to have such a check, it will need to occur when the `PassManager` is instantiated (or rather, when `enable_ir_printing` is called). Since the IR printing is done via the trace infrastructure, we can simply emit a warning diagnostic when a pass name is not registered, rather than raising an error.